Sources/Core/Shared/Operation.swift

```diff
@@ -204,14 +185,60 @@ public class Operation: NSOperation {
     */
     public var log: LoggerType {
         get {
-            _log.operationName = operationName
-            return _log
+            let operationName = self.operationName
+            return _log.read { _LoggerOperationContext(parentLogger: $0, operationName: operationName) }
         }
         set {
-            _log = newValue
+            _log.write { (inout ward: LoggerType) in
+                ward = newValue
+            }
         }
     }
 
+    // MARK: - Initialization
+
+    public override init() {
+        super.init()
+    }
+
+    // MARK: - Disable Automatic Finishing
+
+    /**
+     Ability to override Operation's built-in finishing behavior, if a
+     subclass requires full control over when finish() is called.
+
+     Used for GroupOperation to implement proper .Finished state-handling
+     (only finishing after all child operations have finished).
+
+     The default behavior of Operation is to automatically call finish()
+     when:
+        (a) it's cancelled, whether that occurs:
+            - prior to the Operation starting
+              (in which case, Operation will skip calling execute())
+            - on another thread at the same time that the operation is
+              executing
+        (b) when willExecuteObservers log errors
+
+     To ensure that an Operation subclass does not finish until the
+     subclass calls finish():
+     call `super.init(disableAutomaticFinishing: true)` in the init.
+
+     IMPORTANT: If disableAutomaticFinishing == TRUE, the subclass is
+     responsible for calling finish() in *ALL* cases, including when the
+     operation is cancelled.
+
```

Comment:
Just thinking that here it might be wise to include how a subclass should do this.

Earlier, while developing the `WillCancelObserver` and `DidCancelObserver` it was suggested that `Operation` make its `cancel()` override final. The intention being that subclasses would use these two observers to manage their internal state.